### PR TITLE
added authentication

### PR DESCRIPTION
**Pull request by Mary-Vivian**
# Description

Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context. 
List any dependencies that are required for this change.

Perhaps answering the following questions will help create a good description:
* What is this?
* This PR introduces Single Sign-On (SSO) functionality to the Django backend. SSO allows users to authenticate once and gain access to multiple systems or applications without needing to log in separately for each one. This enhancement integrates SSO capabilities, leveraging an identity provider to manage user authentication and authorization across our platform.

* Why this?
* SSO enhances user experience by reducing login fatigue, improves security through centralized authentication, and simplifies user management.
* How are you doing this? 
* We integrate SSO by configuring Django with the chosen identity provider, updating settings for authentication backends, implementing necessary views and endpoints, and ensuring robust testing and documentation for the new functionality.

Fixes # (issue) [Provide a link to the related Jira Task]

## Type of change

Specify the kind of PR/MR

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update
- [x] Optimization
- [x] Improvement
- [ ] Bug fix

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A - e.g Manually tested user creation
- [ ] Test B

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Maureen-Wangari — APPROVED** [review]

**Review by Yvonne-Martin — APPROVED** [review]

File: landvista/authentication/views.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        request.session['user_info'] = user_info
+        return redirect("register")
+"""
+In this function handle registration"""
+
+def register(request):
+    user_info = request.session.get("user_info")
+    if request.method == "POST":
+        username = user_info["email"]
+        password = User.objects.make_random_password()
```
Comment (Ndichu-shee) [inline landvista/authentication/views.py:89]: why are you making random passwords

File: landvista/users/forms.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from django import forms
```
Comment (Ndichu-shee) [inline landvista/users/forms.py:1]: why are you using forms

File: landvista/users/views.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                user.is_staff = True
+                user.save()
+            messages.success(request, 'User created successfully.')
+            return redirect('admin:index')
+    else:
+        form = UserCreationForm()
+    return render(request, 'users/email_templates.html', {'form': form})
+@csrf_exempt
+@api_view(['POST'])
+def send_test_email(request):
```
Comment (Ndichu-shee) [inline landvista/users/views.py:96]: are you sending emails twice??

Comment (Mary-Vivian) [reply landvista/users/views.py]: No the send-test-email was for testing if the email is being sent but I have removed it and my send-email function is in the utils.py file

File: landvista/authentication/views.py (hunk 1 above)
Comment (Mary-Vivian) [reply landvista/authentication/views.py:89]: This was for generating a temporary password for users

File: landvista/users/forms.py (hunk 2 above)
Comment (Mary-Vivian) [reply landvista/users/forms.py:1]: I was using it if I need to register user in a web interface and not through the api.

**Review by Aber-Racheal — APPROVED** [review]

**Review by Ndichu-shee — APPROVED** [review]

**Pull request merged**